[docs] Reword icons page (#17558)
mui-org/material-ui: docs/src/pages/components/icons/icons.md

<p class="description">Guidance and suggestions for using icons with Material-UI.</p>

Material-UI provides icons support in three ways.

- Standardized Material Design icons are exported as React components from `@material-ui/icons` package (Material Icons).

> - With the `SvgIcon` component as a React wrapper for a custom SVG icon is exported from `@material-ui/core`.

Reviewer: ```suggestion
- With the `SvgIcon` component, a React wrapper for a custom SVG icon, exported from `@material-ui/core`.
```

> - With the `Icon` component as a React wrapper for custom font icons is exported from `@material-ui/core`.

Reviewer: ```suggestion
- With the `Icon` component, used to render custom font icons, exported from `@material-ui/core`.
```

## Material Icons (recommended)

> Material Design has standardized over 1,000 commonly needed icons, each in five different "themes" (see below). For each SVG icon we export the respective React component from the `@material-ui/icons` package. You can search the full list of these icons in [Material Icons](/components/material-icons/).

Reviewer: ```suggestion
Material Design has standardized over 1,000 commonly needed icons, each in five different "themes" (see below). For each SVG icon, we export the respective React component from the `@material-ui/icons` package. You can search the full list of these icons in [Material Icons](/components/material-icons/).
```

> Material Design has standardized over 1,000 commonly needed icons, each in five different "themes" (see below). For each icon we export the respective React component from the `@material-ui/icons` package. Material Icons are based on SVG versions of Material Design icons. You can search the full list of these icons in [Material Icons](/components/material-icons).

Reviewer: ```suggestion
Material Design has standardized over 1,000 commonly needed icons, each in five different "themes" (see below). For each SVG icon we export the respective React component from the `@material-ui/icons` package. You can search the full list of these icons in [Material Icons](/components/material-icons).
```

Not sure if that suggestion was rejected or just lost?

Author: In this case I like my version a bit more. But if you think I should change I will do it :)

Reviewer: It's a bit wordy, no disrespect.

Author: OK. I accept your suggestion then. :)

<p class="description">Guidance and suggestions for using icons with Material-UI.</p>

> 

Author: Not needed and confusing

> 

Author: No need to mention Google I think. It's Material Design

Material-UI provides icons support in three ways.

> 

Author: Provides 3 ways to show icons

- Standardized Material Design icons are exported as React components from `@material-ui/icons` package (Material Icons).

> - `SvgIcon` component as a React wrapper for custom SVG icon is exported from `@material-ui/core`.

Reviewer: ```suggestion
- The `SvgIcon` component as a React wrapper for a custom SVG icon is exported from `@material-ui/core`.
```

> - `Icon` component as a React wrapper for custom Font icons is exported from `@material-ui/core`.

Reviewer: ```suggestion
- The `Icon` component as a React wrapper for custom font icons is exported from `@material-ui/core`.
```

Is it a wrapper?

Author: Not sure. Haven't really touched Font icons. Suggest differently.

Author: What do you say about
- With the `SvgIcon` component...
- With the `Icon` component...

Author: Check latest commit

> 

Author: Move below Material Icons (recommended way)

## Material Icons (recommended)

> Material Design has standardized over 1,000 commonly needed icons. For each icon we export a respective React component from `@material-ui/icons` package. Material Icons are based on SVG icons. You can search the full list of 

Reviewer: ```suggestion
Material Design has standardized over 1,100 commonly needed icons in five different "themes" (see below). For each icon we export the respective React component from  the`@material-ui/icons` package. You can search the full list of 
```

(Arguably many of the provided icons are quite obscure, and some common ones are missing!)

Author: I think it is worth noting that our implementation of Material Design icons (that is Material Icons) is SVG based (as it offers benefits over font icon based). What do you think?

I wouldn't put 1100, because over 1000 is more generic. If they implement 1220 in the next year, we'll have to change it to 1200. What do you think?

Author: What do you say about: ", each in five different "themes" (see below)."

Reviewer: ```suggestion
Material Design has standardized over 1,000 commonly needed icons, each in five different "themes" (see below). For each SVG icon we export the respective React component from  the`@material-ui/icons` package. You can search the full list of 
```

Better?

Author: Check latest commit please.

Make sure you follow the [minimizing bundle size guide](/guides/minimizing-bundle-size/#option-2) before using the second approach.
The configuration of a Babel plugin is encouraged.

Each icon also has a "theme": `Filled` (default), `Outlined`, `Rounded`, `Two tone` and `Sharp`. If you want to import the icon component with a "theme" different than default, append the "theme" name to the icon name. For example `@material-ui/icons/Delete` icon with:

- `Filled` "theme" (default) is exposed as `@material-ui/icons/Delete`,
- `Outlined` "theme" is exposed as `@material-ui/icons/DeleteOutlined`,
- `Rounded` "theme" is exposed as `@material-ui/icons/DeleteRounded`,
- `Two tone` "theme" is exposed as `@material-ui/icons/DeleteTwoTone`,

> - `Sharp` "theme" is exposed as `@material-ui/icons/DeleteSharp`.

Reviewer: Is there a better word than "exposed"?

Author: exposed was used in the previous version of the document. I wasn't sure if it is good, but I tried to use "exported" in other places. I think I missed here. Do you agree "exported" is good?

Reviewer: Probably not technically correct, but I think it's better than "exposed".

> NOTE: Material Design specification names the icons using "C style" naming (for example `delete_forever`, `add_a_photo`), while `@material-ui/icons` exports respective icons using "PascalCase" naming (for example `DeleteForever`, `AddAPhoto`). Also note the three exceptions to the naming rule: `3d_rotation` exported as `ThreeDRotation`, `4k` exported as `FourK`, and `360` exported as `ThreeSixty`.

Reviewer: ```suggestion
NOTE: The Material Design specification names the icons using "snake_case" naming (for example `delete_forever`, `add_a_photo`), while `@material-ui/icons` exports the respective icons using "PascalCase" naming (for example `DeleteForever`, `AddAPhoto`). There are three exceptions to this naming rule: `3d_rotation` exported as `ThreeDRotation`, `4k` exported as `FourK`, and `360` exported as `ThreeSixty`.
```

{{"demo": "pages/components/icons/SvgMaterialIcons.js"}}

## `SvgIcon`

> If you need a custom SVG icon (not available in Material Icons) you should use `SvgIcon` wrapper.

Reviewer: ```suggestion
If you need a custom SVG icon (not available in Material Icons) you should use the `SvgIcon` wrapper.
```

The `SvgIcon` component takes the SVG `path` element as its child and converts it to a React component that displays this SVG icon, 
and allows the icon to be styled and respond to mouse events. SVG elements should be scaled for a 24x24px viewport.

The resulting icon can be used as is,
or included as a child for other Material-UI components that use icons.
By default, an Icon will inherit the current text color.
Optionally, you can set the icon color using one of the theme color properties: `primary`, `secondary`, `action`, `error` & `disabled`.

{{"demo": "pages/components/icons/SvgIcons.js"}}

### Libraries of SVG icons

> There are lots of projects out there that provide SVG icons. For example [https://materialdesignicons.com](https://materialdesignicons.com/) provides over 2,000 icons. For the wanted icon, check the SVG `path` they provide. Put it as the child of `SvgIcon` component and you will get React component for that icon. 

Reviewer: ```suggestion
There are many projects that provide SVG icons. For example [https://materialdesignicons.com](https://materialdesignicons.com/) provides over 2,000 icons. For the wanted icon, copy the SVG `path` they provide, and use it as the child of the `SvgIcon` component. 
```

> NOTE: [mdi-material-ui](https://github.com/TeamWertarbyte/mdi-material-ui) project already wrapped each of these SVG icons with `SvgIcon` component and provides them as React components, so you don't have to do it yourself.

Reviewer: ```suggestion
NOTE: [mdi-material-ui](https://github.com/TeamWertarbyte/mdi-material-ui) has already wrapped each of these SVG icons with the `SvgIcon` component, so you don't have to do it yourself.
```
